streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -166,24 +183,50 @@ public void shouldForwardOldValuesWhenEnabled() {
         store.setFlushListener(cacheFlushListener, true);
         store.put(bytesKey("1"), bytesValue("a"));
         store.flush();
+        assertEquals("a", cacheFlushListener.forwarded.get("1").newValue);
+        assertNull(cacheFlushListener.forwarded.get("1").oldValue);
         store.put(bytesKey("1"), bytesValue("b"));
```

Comment:
I think we should do a third put `store.put(bytesKey("1"), bytesValue("c"));` here and test if old value is `a` and new value is `c` to make sure we return the correct old value -- with 2 puts, it's unclear it it's correct. (If would be incorrect if oldValue would be `b`)

Reply:
Ack.